### PR TITLE
ROX-26530: Add UI analytics for Compliance Reporting

### PR DESCRIPTION
**Pull request by sachaudh**
### Description

Added analytics for Compliance Reporting:
- how many users click to generate a report from the table?
- how many users click to send a report from the table?
- how many users click to generate a report from the details page?
- how many users click to send a report from the details page?
- how many users check the report jobs table?
- how many users toggle to view only their jobs?
- which report run states do users filter by?

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests
- [x] no tests changed

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobs.tsx [hunk 1]
```diff
@@ -76,22 +79,43 @@ function ReportJobs({ scanConfig, isComplianceReportingEnabled }: ReportJobsProp
     const onReportStatesFilterChange = (_checked: boolean, selectedStatus: RunState) => {
         const isStatusIncluded = filteredReportRunStates.includes(selectedStatus);
         if (isStatusIncluded) {
+            const newFilters = ensureReportRunStates(
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobs.tsx:82]: Although off the subject of this contribution, the repetition brings question to mind:
Is it possible to factor out into one assignment, `analyticsTrack` call, and `setSearchFilter` call?

```ts
const newFilters = ensureReportRunStates(
    isStatusIncluded ? filteredReportRunStates.filter(…) : [...filteredReportRunStates, selectedStatus]  
);
// analyticsTrack

@bradr5
// setSearchFilter
```

File: ui/apps/platform/src/Containers/MainPage/Body.tsx [hunk 2]
```diff
@@ -241,11 +241,8 @@ type BodyProps = {
 };
 
 function Body({ hasReadAccess, isFeatureFlagEnabled }: BodyProps): ReactElement {
-    const location = useLocation();
-    const { analyticsPageVisit } = useAnalytics();
-    useEffect(() => {
-        analyticsPageVisit('Page Viewed', '', { path: location.pathname });
-    }, [location, analyticsPageVisit]);
+    useAnalyticsPageTracking();
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/MainPage/Body.tsx:244]: Do we expect to reuse this function?

Unlike reusable `useAnalytics` hook, moving this code reduces visibility of an important side effect.

Comment (bradr5) [reply ui/apps/platform/src/Containers/MainPage/Body.tsx:244]: I agree with Mark here. I might be missing something, but this feels like an additional layer of abstraction for a hook that won't be used anywhere else.

Comment (sachaudh) [reply ui/apps/platform/src/Containers/MainPage/Body.tsx:244]: That's true. We will most likely keep this logic in one place and use it only here. I had originally thought of abstracting it out so I could try to reuse it for specific nested views. After Mark suggested tracking the "intentional" action of clicking the tab to view "All report jobs", I guess we don't need to track specific nested views using a pattern like this. I forgot to come back and change that. I can undo this change.

**Review by pedrottimark — APPROVED** [review]

**Pull request merged**